### ISSUE TITLE
JFormFieldUser broken since 3.5.0

### ISSUE THREAD

**Issue by OctavianC** [joomla/joomla-cms#9593]
#### Steps to reproduce the issue
There are two ways to reproduce this:
- Switch to the Hathor template. I know it's out of date and never updated, but it serves this point for now. Go to Content > Articles > Add New Article. Click on the User icon next to the Created By field. The Modal window shows up (it should be based on MooTools). Click on any user in there and nothing happens.
- I've also noticed that both Protostar and Isis have layout overrides in `/html/layouts/joomla/form/field/user.php`. I do understand that they need overrides to use Bootstrap, but if I remove them I would expect them unstyled but functional. Removing overrides and re-doing the steps above will exhibit the same error.

#### Expected result
Clicking on a user should close the modal window and assign the user to the hidden field.

#### Actual result
Clicking on a user does nothing.

#### System information (as much as possible)
Joomla! 3.5.0 Stable [ Unicorn ] 21-March-2016 22:00 GMT 

#### Additional comments
Seems to be caused by https://github.com/joomla/joomla-cms/pull/5655
I've noticed this part https://github.com/joomla/joomla-cms/blob/staging/layouts/joomla/form/field/user.php#L50 that seems strange to me:
```php
. ($required ? 1 : 0) . '&amp;field={field-user-id}'
```
I'm guessing replacing it with:
```php
. ($required ? 1 : 0) . '&amp;field=' . htmlspecialchars($id, ENT_COMPAT, 'UTF-8')
```
Would do the trick. But I'm not sure if I'm missing something.

**Comment by wilsonge** [comment]
@dgt41

**Comment by OctavianC** [comment]
I'm not sure how jSelectUser is supposed to be triggered by MooTools modals since the function is not called `onclick` in `administrator/components/com_users/views/users/tmpl/modal.php`.

I did some tests and can try a PR, but I'm not sure if I entirely understand the point of `fielduser.js` - it will only inject the click event in Bootstrap modals.

TLDR: My 2 cents: inject the `onclick` event on page load so it works across all modals and call `jSelectUser()` because why not, since it does what it's supposed to.

**Comment by dgt41** [comment]
@OctavianC The main difference between the mootools code and the bootstrap is that the later will allow to initialise fields dynamically (this was a prerequisite for the next version of repeatable forms), so in that sense the id is kinda irrelevant where the mootools script (and that kept for B/C) expects the id to be unique etc...

Few of the aims for all the fields should be: 
- to make them repeatable friendly,
- discard the inline scripts
- use data attributes
- use native javascript where possible

**Comment by brianteeman** [comment]
Closed as we have a PR #9659 for testing<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9593">issues.joomla.org/joomla-cms/9593</a>.</sub>

**Comment by molszews** [comment]
After upgrading to 3.6.0 selecting users in external components seems to be broken. Clicking on one in e.g. Akeeba Subscription causes following error in console: `window.parent.jSelectUser is not a function`
After adding following line
    JHtml::script('jui/fielduser.min.js', false, true, false, false, true);
there is no error in console, but clicking on name still gives no effect in that particular component.

Does changes made in 3.5.0 require components to be adjusted in order to work with forms, specifically user selection?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9593">issues.joomla.org/joomla-cms/9593</a>.</sub>

**Comment by marclight** [comment]
@molszews were you able to find a fix?  I have the same problem.

**Comment by dgt41** [comment]
@marclight update to 3.6.4

**Comment by marclight** [comment]
@dgt41 I'm at 3.6.4 and I'm having the same problem as @molszews with Akeeba Subscriptions

**Comment by brianteeman** [comment]
Have you asked akeeba for support?

**Comment by marclight** [comment]
@brianteeman I wish, they don't support Akeeba Subscriptions anymore.  Google sent me here because of what @molszews posted.  Was hoping he might got it working.

**Comment by dgt41** [comment]
@marclight do you have the code from https://github.com/joomla/joomla-cms/pull/10791 to your site?

**Comment by molszews** [comment]
@marclight still no luck

**Comment by OctavianC** [comment]
This is not an issue with Joomla! - I've debugged the code and found that Akeeba Subscriptions does not use the JFormFieldUser field - it uses its own HTML, so that's why it isn't working.

**Comment by marclight** [comment]
@OctavianC THanks for the info.

@molszews I'll put a dev on it and if we make it work, I'll let you know.

**Comment by OctavianC** [comment]
Workaround:
Open `/administrator/components/com_akeebasubs/views/user/tmpl/form.php`
After:
```html
<script type="text/javascript">
```
Add:
```javascript
function jSelectUser(element)
{
	jSelectUser_userid(jQuery(element).data('user-value'), jQuery(element).data('user-name'));
}
```

**Comment by marclight** [comment]
@OctavianC I still get the following:

Uncaught TypeError: window.parent.jSelectUser is not a function at HTMLAnchorElement.onclick

**Comment by OctavianC** [comment]
Just tested it and it works, perhaps you need to add it in another view that uses modals (found multiple - user, subscription, coupon etc). Sorry, we're going off-topic.

**Comment by marclight** [comment]
@OctavianC Added the function in all the views and it now works!  Thanks a million ans sorry too for going off topic :)
